compiler/packages/babel-plugin-react-compiler/src/Inference/InferReferenceEffects.ts

```diff
@@ -1287,20 +1304,18 @@ function inferBlock(
         break;
       }
       case 'PropertyStore': {
-        const effect =
-          state.kind(instrValue.object).kind === ValueKind.Context
-            ? Effect.ConditionallyMutate
-            : Effect.Capture;
         state.referenceAndRecordEffects(
           freezeActions,
           instrValue.value,
-          effect,
+          Effect.Capture,
           ValueReason.Other,
         );
         state.referenceAndRecordEffects(
           freezeActions,
           instrValue.object,
-          Effect.Store,
+          typeof instrValue.property === 'string'
+            ? Effect.Store
+            : Effect.Mutate,
```

Comment:
I need to follow-up on this: this is handling some cases i saw with array access, but i think i've subsequently fixed them by changing the rvalue to be capture and not mutate. I'll double-check.

Reply:
See later PRs in the stack, i confirmed that the effect here needs to be Store (if known object type) or else Mutate, regardless of the property kind